.github/workflows/multi-arch-test-build.yml

```diff
@@ -35,9 +35,17 @@ jobs:
       - name: Determine changed packages
         run: |
           # only detect packages with changed Makefiles
-          PACKAGES="$(git diff --diff-filter=d --name-only origin/master \
-            | grep -E 'Makefile$|test.sh$' | grep -Ev '/files/|/src/' \
-            | awk -F/ '{ print $(NF-1) }' | tr '\n' ' ')"
+          PKG_ROOTS=$(find -name Makefile | sed -e 's@./\(.*\)/Makefile@\1/@')
+          CHANGES=$(git diff --diff-filter=d --name-only origin/master)
+
+          for ROOT in $PKG_ROOTS; do
+            for CHANGE in $CHANGES; do
```

Comment:
Shouldn't we swap these as it's likely always only a single package that changes?

Reply:
Yeah, then the de-dup logic will need some work - I'll take a look

Reply:
I don't think it's really worth optimizing this - even in CI this whole thing is done in 1 second or less. Take a look at the CI logs - temp commit prints out `date` before and after this check.